File: doller.go
package money

type doller struct {
	value
}

func NewDoller(amount int) *doller {
	doller := new(doller)
	doller.amount = amount
	return doller
}

func (d *doller) times(multiplier int) value {
	return NewDoller(d.amount * multiplier).value
}

File: src/Money/Doller.php
<?php

namespace Money;

class Doller
{
    private $Amount;
    
    public function __construct(int $amount)
    {
        $this->Amount = $amount;
    }

    public function times(int $multiplier)
    {
        return new Doller($this->Amount * $multiplier);
    }

    public function equals(Doller $doller) : bool
    {
        return $this->Amount == $doller->Amount;
    }
}